pkg/cmd/server/origin/master_config.go

```diff
@@ -670,6 +671,21 @@ func newAuthenticator(config configapi.MasterConfig, restOptionsGetter restoptio
 		},
 	}
 
+	//	if we have a front proxy providing authentication configuration, wire it up and it should come first
+	if config.AuthConfig.RequestHeader != nil {
+		requestHeaderAuthenticator, err := headerrequest.NewSecure(
+			config.AuthConfig.RequestHeader.ClientCA,
+			config.AuthConfig.RequestHeader.ClientCommonNames,
+			config.AuthConfig.RequestHeader.UsernameHeaders,
+			config.AuthConfig.RequestHeader.GroupHeaders,
+			config.AuthConfig.RequestHeader.ExtraHeaderPrefixes,
+		)
+		if err != nil {
+			return nil, err
+		}
+		ret.Handlers = append([]authenticator.Request{requestHeaderAuthenticator}, ret.Handlers...)
```

Comment:
Should be first inside the list of handlers in the group adder, right? Why not just add it to `authenticators` first at the top of the method?